internal/resources/resource_lb_virtual_servers.go

```diff
@@ -15,87 +17,123 @@ import (
-				Required:    true,
+				Optional:    true,
 				Description: "vip_address of Network loadbalancer virtual server",
-				ForceNew:    true,
 			},
 			"vip_port": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "vip_port of network loadbalancer virtual server",
 			},
 			"pool": {
 				Type:        schema.TypeInt,
-				Required:    true,
+				Optional:    true,
 				Description: "pool of Network loadbalancer virtual server",
 			},
-			"ssl_cert": {
-				Type:        schema.TypeInt,
-				Required:    true,
-				Description: "ssl_cert of Network loadbalancer virtual server",
+			"vip_host_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "vip_host_name of Network loadbalancer virtual server",
+			},
+			"type": {
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validations.StringInSlice([]string{"http", "tcp", "udp"}, false),
+				Optional:         true,
+				Default:          "http",
+				Description:      "vip protocol of Network loadbalancer virtual server",
 			},
+			"tcp_application_profile":  schemas.TcpAppProfileSchema(),
+			"udp_application_profile":  schemas.UdpAppProfileSchema(),
+			"http_application_profile": schemas.HttpAppProfileSchema(),
+			"persistence": {
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validations.StringInSlice([]string{"SOURCE_IP", "COOKIE"}, false),
+				Optional:         true,
+				Description:      "vip_host_name of Network loadbalancer virtual server",
+			},
+			"cookie_persistence_profile":   schemas.CookiePersProfileSchema(),
+			"sourceip_persistence_profile": schemas.SourceipPersProfileSchema(),
 			"ssl_server_cert": {
 				Type:        schema.TypeInt,
-				Required:    true,
+				Optional:    true,
 				Description: "ssl_server_cert of the Network loadbalancer virtual server",
 			},
-			"config": {
+			"ssl_server_config": {
 				Type:        schema.TypeList,
-				Required:    true,
+				Optional:    true,
 				Description: "virtual server Configuration",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"persistence": {
-							Type:             schema.TypeString,
-							ValidateDiagFunc: validations.StringInSlice([]string{"SOURCE_IP", "COOKIE", "DISBALED"}, false),
-							Required:         true,
-							Description:      "Network Loadbalancer Supported values are `SOURCE_IP`,`COOKIE`, `DISBALED`"},
-						"persistence_profile": {
+						"ssl_server_profile": {
 							Type:        schema.TypeInt,
 							Optional:    true,
-							Description: "persistence_profile of virtual server Configuration",
-						},
-						"application_profile": {
-							Type:        schema.TypeInt,
-							Required:    true,
-							Description: "application_profile of virtual server Configuration",
+							Description: "Network Loadbalancer Supported values are `SOURCE_IP`,`COOKIE`, `DISBALED`",
 						},
+					},
+				},
+			},
+			"ssl_client_cert": {
```

Comment:
why is this Int ? what Integer doesnt user need to provide ?

Reply:
that is the ID from SSL_CA_certs so added the datasource to it you can refer the example for vs resource

Reply:
u should give a ref to that here, how will the user know how to get this data ?